src/Codegen/Builders/Fields/FieldBuilder.hack

```diff
@@ -14,6 +14,7 @@ abstract class FieldBuilder {
     abstract const type TField as shape(
         'name' => string,
         'output_type' => shape('type' => string, ?'needs_await' => bool),
+        'directives' => vec<string>,
```

Comment:
In the spec directives can take arguments so you might want to future proof this and make it dict<string, mixed>. For sure you also don't want to double-up on the directives. And you'll also want key-based lookup instead of C\contains($vec, $thing) which is O(n) linear search.